src/flows/service_functions/communication.py

```diff
@@ -229,7 +236,10 @@ def send_room_state(
     if room_state is None:
         return
     payload = build_room_state_payload(caller_state, room_state)
-    cast(Any, caller_state.obj).msg(room_state=((), payload))
+    if isinstance(caller_state.obj, MsgCapable):
+        caller_state.obj.msg(room_state=((), payload))
+    elif hasattr(caller_state.obj, 'msg'):
+        caller_state.obj.msg(room_state=((), payload))
```

Comment:
obj should be asserted to be an instance of ObjectDB (or its typeclass subclasses in our typeclasses/ dir, which are set by the metaclass when instantiated), which has `msg` defined as a method. the checks are not necessary.